### PR TITLE
[GPU] Fix Loop to cover the case where memory is null at the first iteration

### PR DESCRIPTION
**Pull request by davidsnam-intel**
### Details:
 - Handle the case where from_mem is null at the first iteration 

### Tickets:
 - 141652

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/loop.cpp [hunk 1]
```diff
@@ -526,6 +533,13 @@ void loop_inst::preprocess_backedge_memory() {
         OPENVINO_ASSERT(!input_map_ptrs.empty(), id(), " has no input_mapping for backedged input");
         auto& external_id = input_map_ptrs.front()->external_id;
         initial_mem = get_external_memory(external_id.pid, external_id.idx);
+        auto initial_layout = get_external_output_layout(external_id.pid, external_id.idx);
```
Comment (ahnyoung-paul) [inline src/plugins/intel_gpu/src/graph/loop.cpp:536]: please add description for this code

File: src/plugins/intel_gpu/src/graph/loop.cpp [hunk 2]
```diff
@@ -985,7 +999,15 @@ int64_t loop_inst::get_num_iterations() {
 void loop_inst::set_memory_in_body_network(cldnn::network::ptr body_network,
                 const std::shared_ptr<cldnn::primitive_inst>& inst, memory::ptr mem) {
     if (inst->is_input()) {
-        body_network->set_input_data(inst->id(), mem);
+        memory::ptr updated_mem = mem;
```
Comment (ahnyoung-paul) [inline src/plugins/intel_gpu/src/graph/loop.cpp:1002]: please add description for this code

**Comment by ahnyoung-paul** [comment]
Please add unit test for this code change

File: src/plugins/intel_gpu/src/graph/loop.cpp (hunk 1 above)
Comment (davidsnam-intel) [reply src/plugins/intel_gpu/src/graph/loop.cpp:538]: description has been added

File: src/plugins/intel_gpu/src/graph/loop.cpp (hunk 2 above)
Comment (davidsnam-intel) [reply src/plugins/intel_gpu/src/graph/loop.cpp:1006]: description has been added

**Review by yeonbok — APPROVED** [review]

**Pull request merged**